### DISCUSSION TITLE
Question - Which version of Firefox is Floorp current in?

### DISCUSSION THREAD

**Discussion by mateus3c** [Floorp-Projects/Floorp#620] [Q&A]
I have heard that the newest Firefox version (119+) got way faster than what we were used to, so I was wondering if Floorp is up-to-date? Sometimes I feel like Floorp is significantly slower than other browsers, an extensive benchmark made by Diolinux showed pretty bad statistics... ( https://www.youtube.com/watch?v=Jyv3Aj5oGsU&t=349s )

**Comment by CutterKnife** [comment]
Floorp is based on [Firefox ESR](https://www.mozilla.org/en-US/firefox/enterprise/). This is the version that receives major updates less frequently than usual, but security fixes are applied as usual.
The next major ESR update will be 128.

**Reply by mateus3c** [reply]
alright, thank you for explaining! do you know if theres any estimated release date? I'm looking foward to see Floorp running faster 👀

**Reply by CutterKnife** [reply]
Hmmm, undecided at this time.
If it’s the same as this year, it will be around summer 2024.

Major ESR updates have a transition period, giving users of the ESR version a few more months.
Please see [here](https://whattrainisitnow.com/calendar).

**Reply by RRichardsAsmadi** [reply]
Okay just wanting to double check we're atleast past version 91 as a base or have otherwise applied a patch for CVE-2023-4863  yeah? i'd imagine soo given it's december now and the patch was pushed in september but I wanted to double check.

**Reply by CutterKnife** [reply]
You can check the current version of Floorp and the corresponding ESR version at about:preferences#general.

**Reply by e-t-l** [reply]
How quickly does Floorp publish an update after FF ESR gets updated (e.g. with security patches)? How quickly does Floorp get updated on the Winget repo? I know Winget can sometimes lag with getting the latest versions of stuff)

**Reply by CutterKnife** [reply]
Regular security fixes are distributed earlier than upstream and urgent security fixes are [distributed](https://github.com/Floorp-Projects/Floorp/releases/tag/v11.19.1) within two days of the [upstream release](https://github.com/advisories/GHSA-hm3j-qgpw-pj98).
Winget versions are usually [distributed](https://github.com/microsoft/winget-pkgs/pull/180708) within a few days of the Floorp release.

**Reply by e-t-l** [reply]
That's a great answer, thank you. Now I'm curious, what does "earlier than upstream" mean?

**Reply by CutterKnife** [reply]
AFAIK, Firefox ESR (which Floorp 11 is based on) has a “waiting for shipping” phase before it is actually released; Floorp incorporates this code when it updates, resulting in an earlier release than Firefox.
Urgent security fixes do not go through this release-waiting phase, so there is a delay in getting them into Floorp.
